### PR TITLE
changes for change log detail description

### PR DESCRIPTION
**Pull request by kth13**
## What does this pull request do?

change log detail api transforms the data from the repository to be presented in the user interface

## What is the intent behind these changes?

The detailed change log view requires some of the change log information to be presented in a human understandable format.

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/AmendReferralServiceTest.kt [hunk 1]
```diff
@@ -256,18 +273,140 @@ class AmendReferralServiceTest @Autowired constructor(
       newVal = ReferralAmendmentDetails(listOf("hearing aid")),
       changedBy = someoneElse
     )
-    changeLogFactory.create(
+    val changelog2 = changeLogFactory.create(
```
Comment (markmoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/AmendReferralServiceTest.kt:276]: use change log factory for consistency

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/controller/AmendReferralControllerTest.kt [hunk 2]
```diff
@@ -256,15 +257,109 @@ internal class AmendReferralControllerTest {
...
+      val returnedChangeLogObject = amendReferralController.getChangelogDetails(changelog.id, token)
+
+      assertThat(returnedChangeLogObject.changelogId).isEqualTo(changelog.id)
+      assertThat(returnedChangeLogObject.referralId).isEqualTo(changelog.referralId)
+    }
+
+    @Test
+    fun `getChangelogDetails returns the complexity title for complexity level changelog entry`() {
+
+      val changelog = Changelog(
```
Comment (markmoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/controller/AmendReferralControllerTest.kt:284]: use change log factory for consistency

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/controller/AmendReferralControllerTest.kt [hunk 3]
```diff
@@ -256,15 +257,109 @@ internal class AmendReferralControllerTest {
...
+      assertThat(returnedChangeLogObject.changelogId).isEqualTo(changelog.id)
+      assertThat(returnedChangeLogObject.referralId).isEqualTo(changelog.referralId)
+      assertThat(returnedChangeLogObject.oldValue).containsExactly("OLDTITLE")
+      assertThat(returnedChangeLogObject.newValue).containsExactly("NEWTITLE")
+    }
+
+    @Test
+    fun `getChangelogDetails returns the desired outcomes description for desired outcomes changelog entry`() {
+
+      val changelog = Changelog(
```
Comment (markmoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/controller/AmendReferralControllerTest.kt:313]: use change log factory for consistency

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/controller/AmendReferralControllerTest.kt [hunk 4]
```diff
@@ -256,15 +257,109 @@ internal class AmendReferralControllerTest {
...
+      assertThat(returnedChangeLogObject.changelogId).isEqualTo(changelog.id)
+      assertThat(returnedChangeLogObject.referralId).isEqualTo(changelog.referralId)
+      assertThat(returnedChangeLogObject.oldValue).containsExactlyInAnyOrder("desc1", "desc2")
+      assertThat(returnedChangeLogObject.newValue).containsExactly("desc3")
+    }
+
+    @Test
+    fun `getChangelogDetails returns the amended description for interpreter required changelog entry`() {
+
+      val changelog = Changelog(
```
Comment (markmoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/controller/AmendReferralControllerTest.kt:346]: use change log factory for consistency

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/controller/AmendReferralControllerTest.kt (hunk 2 above)
Comment (kth13) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/controller/AmendReferralControllerTest.kt]: Updated.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/controller/AmendReferralControllerTest.kt (hunk 3 above)
Comment (kth13) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/controller/AmendReferralControllerTest.kt]: Updated.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/controller/AmendReferralControllerTest.kt (hunk 4 above)
Comment (kth13) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/controller/AmendReferralControllerTest.kt]: Updated.

**Review by markmoj — APPROVED** [review]
fine

**Pull request merged**